x-pack/plugins/threat_intelligence/public/modules/indicators/components/indicator_field/indicator_field.tsx

```diff
@@ -20,19 +21,21 @@ export interface IndicatorFieldProps {
    * The field to get the indicator's value for.
    */
   field: string;
-  /**
-   * An object to know what type the field is ('file', 'date', ...).
-   */
-  fieldTypesMap: { [id: string]: string };
 }
 
 /**
  * Takes an indicator object, a field and a field => type object to returns the correct value to display.
  * @returns If the type is a 'date', returns the {@link DateFormatter} component, else returns the value or {@link EMPTY_VALUE}.
  */
-export const IndicatorField: VFC<IndicatorFieldProps> = ({ indicator, field, fieldTypesMap }) => {
+export const IndicatorField: VFC<IndicatorFieldProps> = ({ indicator, field }) => {
+  const indicatorsTableContext = useContext(IndicatorsTableContext);
```

Comment:
what about making a pure component that receives `indicator`, `field`, and `fieldType` and then also a wrapper that gets the `fieldType` from the table context? I can totally see that we might potentially want to display an indicator field value outside of the table context.

Reply:
you are right, this should not be nested inside the tablecontext, I will avoid this somehow.

Reply:
I will move this value to a higher level context wrapping the entire plugin then. My intention was to reduce the amount of passing this as a prop around all the components.

Reply:
Please, take a look now. It is much simpler now and the mapping can be used anywhere on the page, not just inside the table.